search/api.py

```diff
@@ -0,0 +1,207 @@
+"""
+Functions for search
+"""
+from itertools import islice
+import logging
+
+from django.conf import settings
+from elasticsearch.helpers import bulk
+from elasticsearch.exceptions import NotFoundError
+from elasticsearch_dsl import Mapping
+from elasticsearch_dsl.connections import connections
+
+from profiles.models import Profile
+from profiles.serializers import ProfileSerializer
+from search.exceptions import ReindexException
+
+log = logging.getLogger(__name__)
+
+
+PROFILE_DOC_TYPE = 'profile'
+
+DOC_TYPES = (PROFILE_DOC_TYPE, )
+
+_CONN = None
+# When we create the connection, check to make sure all appropriate mappings exist
+_CONN_VERIFIED = False
```

Comment:
I'm not sure this gives us much. We'll still need some global to store the `ESConnection`, and the helper function `verify` would only be called in one place. `get_conn` already encapsulates the verification so that it happens when a connection is requested